.travis.yml

```diff
@@ -4,7 +4,143 @@ jdk:
   - openjdk8
 script:
 - xvfb-run mvn verify -B -Pjacoco
-- bash deployment_check.sh

```

Comment:
I do like having a separate script for the test. It keeps the yaml file clean. Please do not run this for every build like you are doing here. Use instead a conditional build for your branch. https://docs.travis-ci.com/user/conditional-builds-stages-jobs/#conditional-builds. You would need to maintain a branch mirroring the master that would execute this check and could be triggered by a cron job.

Another approach and with no maintenance could be to detect if the build was activated by a cron job. This approach would use any branch actually and does not need to keep updated a separate branch. See https://docs.travis-ci.com/user/cron-jobs/#detecting-builds-triggered-by-cron